src/osemgrep/cli_scan/Rule_filtering.mli

```diff
@@ -1,11 +1,9 @@
-type product = SAST | SCA | Secrets | Interfile [@@deriving show]
```

Comment:
I think this is intentional so we don't need to use polymorphic variants everywhere.

Reply:
No, I introduced this type for this file, pad didn't like it and wanted me to use what the type now is, but that type didnt cover interfile, but I realized in this PR that's fine

Reply:
Oh sorry I was also thinking this was in Product.ml.

Reply:
thanks! me like removing it.